### PR TITLE
move packaging CmdLets to separate module

### PR DESCRIPTION
**Pull request by TravisEz13**
This is to make it easier to work on packaging changes in the future.

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -0,0 +1,599 @@
...
+            }
+
+            $Arguments = @{
+                ProductNameSuffix = $NameSuffix
+                ProductSourcePath = $Source
+                ProductVersion = $Version
+                AssetsPath = "$PSScriptRoot\..\..\assets"
+                LicenseFilePath = "$PSScriptRoot\..\..\assets\license.rtf"
+                # Product Guid needs to be unique for every PowerShell version to allow SxS install
+                ProductGuid = [Guid]::NewGuid();
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: Maybe use New-Guid cmdlet here. Not a blocker though.

**Review by adityapatwardhan — APPROVED** [review]
LGTM other than minor comment.

**Review by daxian-dbw — COMMENTED** [review]
2 minor comments

File: build.psm1 [hunk 2]
```diff
@@ -1,62 +1,74 @@
...
+        $Runtime = [System.Runtime.InteropServices.RuntimeInformation]
+        $OSPlatform = [System.Runtime.InteropServices.OSPlatform]
+
+        $environment += @{'IsCoreCLR' = $true}
+        $environment += @{'IsLinux' = $Runtime::IsOSPlatform($OSPlatform::Linux)}
+        $environment += @{'IsOSX' = $Runtime::IsOSPlatform($OSPlatform::OSX)}
+        $environment += @{'IsWindows' = $Runtime::IsOSPlatform($OSPlatform::Windows)}
+    } catch {
+        # If these are already set, then they're read-only and we're done
+        try {
```
Comment (TravisEz13) [inline build.psm1]: This is still needed as the types do not exist on FullCLR

Comment (TravisEz13) [inline build.psm1]: See https://ci.appveyor.com/project/PowerShell/powershell/build/6.0.0-beta.4-4127

**Comment by TravisEz13** [comment]
Merging as PR comments have been addressed.  I'll be submitting additional changes, so please feel free to add more comments.

File: build.psm1 (hunk 2 above)
Comment (daxian-dbw) [inline build.psm1]: I mean the inner `try { ... } catch { }` block in the outter `catch { ... }` block and the comment `# If these are already set, then they're read-only and we're done`

Comment (TravisEz13) [inline build.psm1]: submitted #4328 to fix this

**Pull request merged**